dog_classifier/evaluate/evaluate_training.py

```diff
@@ -0,0 +1,277 @@
+    """defines a multiclass probability in the case that the output of the cnn
+    cannot be interpreted as a probability. Also plots for a given label
+    
+    :param Y_pred: model prediction of the test data
+    :param Y_test: test output data
+    :param label: number (or rather name?) of class, which should be examined
+    :param fname: name of plot
+    """
+    print("plot multiclass probability")
+    n_cls = len(Y_pred[0])
+
+    Y_prob = []
+    for i in range(len(Y_pred)):
+        numerator = Y_pred[i, label]
+        denominator = 0.0
+        for idx in range(n_cls):
+            denominator += Y_pred[i, idx]
+
+        Y_prob.append(numerator/denominator)
+
+    Y_pred_prob = np.asarray(Y_prob)
+
+    # Convert validation observations to one hot vectors
+    Y_true = np.argmax(Y_test, axis=1)
+
+    # plt with given label
+    plt.hist(Y_pred_prob[Y_true == label], alpha=0.5, color='red',
+             bins=10, log=True, histtype='step')
+    plt.hist(Y_pred_prob[Y_true != label], alpha=0.5, color='blue',
+             bins=10, log=True, histtype='step')
+    # plt.legend(['digit == {}'.format(label), 'digit != {}'.format(label)],
+    #            loc='upper right')
+    plt.legend([r'class {} matching'.format(label),
+               r"class {} not matching".format(label)], loc='best')
+    plt.xlabel(r'Probability of being class {}'.format(label))
+    plt.ylabel(r'Number of entries')
+    plt.savefig("build/{}.pdf".format(fname))
+    plt.clf()
+
+
+def plot_confusion_matrix(cm, classes, fname,
+                          normalize=True,
+                          title='Confusion matrix',
+                          cmap=plt.cm.Blues
+                          ):
+    """
+    This function prints and plots the confusion matrix.
+    Normalization can be applied by setting `normalize=True`.
+    Taken straight vom SKLEARN.
+
+    :param cm: confusion matrix generated by sklearn
+    :param classes: (name of?) classes to plot
+    :param fname: name of plot
+    """
+    print("plot confusion matrix")
+    plt.imshow(cm, interpolation='nearest', cmap=cmap)
+    plt.title(title)
+    plt.colorbar()
+    tick_marks = np.arange(len(classes))
+    plt.xticks(tick_marks, classes, rotation=45)
+    plt.yticks(tick_marks, classes)
+
+    if normalize:
+        cm = cm.astype('float') / cm.sum(axis=1)[:, np.newaxis]
+
+    # Loop over data dimensions and create text annotations.
+    fmt = '.2f' if normalize else 'd'
+    thresh = cm.max() / 2.
+    for i, j in itertools.product(range(cm.shape[0]), range(cm.shape[1])):
+        plt.text(j, i, format(cm[i, j], fmt),
+                 horizontalalignment="center",
+                 color="white" if cm[i, j] > thresh else "black")
+
+    plt.tight_layout()
+    plt.ylabel('True label')
+    plt.xlabel('Predicted label')
+    plt.savefig("build/{}.pdf".format(fname))
+    plt.clf()
+
+
+def display_errors(n, Y_cls, Y_true, Y_pred, X_test,
+                   height,
```

Comment:
Wieso schreibst du nicht mehr als ein Argument in die Zeile?

Reply:
Das war ein Überbleibsel, ich machs weg

Reply:
Ahh okay, perfekt :)